pkg/types/encrypted_private_key.go

```diff
@@ -112,7 +121,37 @@ func (e *EncryptedPrivateKey) PublicKey(password *memguard.LockedBuffer, salt, n
 	}, nil
 }
 
-func PrivateKey(password *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) (*memguard.LockedBuffer, error) {
+// PrivateKeyTextInClear returns the private key in clear. Password is destroyed.
+func (e *EncryptedPrivateKey) PrivateKeyTextInClear(password *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) (*memguard.LockedBuffer, error) {
+	privateKeyInClear, err := privateKey(password, salt, nonce, encryptedKey)
+	if err != nil {
+		return nil, fmt.Errorf("PrivateKeyTextInClear: %w", err)
+	}
+
+	seed := ed25519.PrivateKey(privateKeyInClear.Bytes()).Seed()
+	privateKeyInClear.Destroy()
+
+	seedBuffer := memguard.NewBufferFromBytes(seed)
+
+	privateKey := e.Kind.Prefix() + base58.CheckEncode(seedBuffer.Bytes(), e.Version)
+	privateKeyBuffer := memguard.NewBufferFromBytes([]byte(privateKey))
+
+	return privateKeyBuffer, nil
+}
+
+// PasswordIsValid returns true if the password is valid for the account. It destroys the password.
+func (e *EncryptedPrivateKey) PasswordIsValid(password *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) bool {
+	privateKeyInClear, err := privateKey(password, salt, nonce, e.Data)
+	if err != nil {
+		return false
+	}
```

Comment:
The logic of this function is weird. If the salt or the nonce is wrong, but the password is ok, you will also get an error. and the function PasswordIsValid will return false...
Furthermore, are you sure that except from an invalid salt, nonce or password, no other error are returned ?